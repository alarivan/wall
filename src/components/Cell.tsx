import styled from 'styled-components';

interface Props {
  readonly cellColor?: string;
  readonly border?: boolean;
}

function getCellColor({ cellColor }: Props): string {
  return cellColor ? cellColor : 'transparent';
}

function getBorder(props: Props): string {
  return props.border ? 'black' : getCellColor(props);
}

const StyledCell = styled.div<Props>`
  background: ${getCellColor};
  border-style: solid;
  border-color: ${getBorder};
  border-width: 0 1px 1px 0;
  transition: transform 100ms ease-in-out;
  cursor: pointer;

  :hover {
    border-width: 1px;
    transform: scale(1.1);
  }
`;

export default StyledCell;
