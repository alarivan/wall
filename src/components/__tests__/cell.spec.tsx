import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Cell from '../Cell';

afterEach(cleanup);

describe('Cell', () => {
  it('renders correctly', () => {
    const { asFragment, container } = render(<Cell cellColor={null} />);

    expect(container.firstChild).toHaveStyleRule('background', 'transparent');
    expect(container.firstChild).toHaveStyleRule('border-color', 'transparent');
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    const props = { cellColor: 'red', border: true };
    const { asFragment, container } = render(<Cell {...props} />);

    expect(container.firstChild).toHaveStyleRule('background', 'red');
    expect(container.firstChild).toHaveStyleRule('border-color', 'black');
    expect(asFragment()).toMatchSnapshot();
  });
});
