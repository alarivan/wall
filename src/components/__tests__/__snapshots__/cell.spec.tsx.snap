// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  background: transparent;
  border-style: solid;
  border-color: transparent;
  border-width: 0 1px 1px 0;
  -webkit-transition: -webkit-transform 100ms ease-in-out;
  -webkit-transition: transform 100ms ease-in-out;
  transition: transform 100ms ease-in-out;
  cursor: pointer;
}

.c0:hover {
  border-width: 1px;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`Cell renders correctly with props 1`] = `
<DocumentFragment>
  .c0 {
  background: red;
  border-style: solid;
  border-color: black;
  border-width: 0 1px 1px 0;
  -webkit-transition: -webkit-transform 100ms ease-in-out;
  -webkit-transition: transform 100ms ease-in-out;
  transition: transform 100ms ease-in-out;
  cursor: pointer;
}

.c0:hover {
  border-width: 1px;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<div
    class="c0"
  />
</DocumentFragment>
`;
